core/auth/src/main/java/software/amazon/awssdk/auth/credentials/HttpCredentialsProvider.java

```diff
@@ -98,11 +103,6 @@ protected boolean isLocalCredentialLoadingDisabled() {
                                 .build();
         } catch (SdkClientException e) {
             throw e;
-        } catch (JsonMappingException e) {
-            throw SdkClientException.builder()
-                                    .message("Unable to parse response returned from service endpoint.")
-                                    .cause(e)
-                                    .build();
```

Comment:
The same failure scenario (invalid response JSON) is now thrown as an UncheckedIOException (caught below).